twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -0,0 +1,155 @@
+load("//scala:scala.bzl", "scala_mvn_artifact", "scala_library", "write_manifest", "implicit_deps")
+
+# TODO is it possible to share scala-library with scala.bzl?
+def twitter_scrooge():
+  native.maven_jar(
+    name = "libthrift",
+    artifact = "org.apache.thrift:libthrift:0.8.0",
+    sha1 = "2203b4df04943f4d52c53b9608cef60c08786ef2",
+  )
+  native.maven_jar(
+    name = "scrooge_core",
+    artifact = scala_mvn_artifact("com.twitter:scrooge-core:4.6.0"),
+    sha1 = "84b86c2e082aba6e0c780b3c76281703b891a2c8",
+  )
+
+  #scrooge-generator related dependencies
+  native.maven_jar(
+    name = "scrooge_generator",
+    artifact = scala_mvn_artifact("com.twitter:scrooge-generator:4.6.0"),
+    sha1 = "cacf72eedeb5309ca02b2d8325c587198ecaac82",
+  )
+  native.maven_jar(
+    name = "scopt",
+    artifact = scala_mvn_artifact("com.github.scopt:scopt:3.3.0"),
+    sha1 = "9e6fb062b2f12a7945b9f38ee54784aee7edde20",
+  )
+  native.maven_jar(
+    name = "util_core",
+    artifact = scala_mvn_artifact("com.twitter:util-core:6.33.0"),
+    sha1 = "bb49fa66a3ca9b7db8cd764d0b26ce498bbccc83",
+  )
+  native.maven_jar(
+    name = "util_logging",
+    artifact = scala_mvn_artifact("com.twitter:util-logging:6.33.0"),
+    sha1 = "3d28e46f8ee3b7ad1b98a51b98089fc01c9755dd",
+  )
+  native.maven_jar(
+    name = "scala_parser_combinators",
+    artifact = scala_mvn_artifact("org.scala-lang.modules:scala-parser-combinators:1.0.4"),
+    sha1 = "7369d653bcfa95d321994660477a4d7e81d7f490",
+  )
+
+def _gen_scrooge_srcjar_impl(ctx):
+  #TODO would there be a case in which we want/need to allow more jars here?
+  transitive_cjars = [ctx.files._scrooge_core, ctx.files._libthrift, ctx.files._scalalib]
+  transitive_cjars = [file for files in transitive_cjars for file in files]
+
+  cjars = ctx.files._scrooge_generator + \
+          ctx.files._scopt + \
+          ctx.files._util_core + \
+          ctx.files._util_logging + \
+          ctx.files._scala_parser_combinators + \
+          transitive_cjars
+  write_manifest(ctx) #TODO do we need something like this?
+
+  #TODO better understand the use of gen-source-map
+  #TODO what imports do we need?
+  cmd = """
+rm -rf {out}_tmp
+set -e
+{java} -classpath "{jars}" com.twitter.scrooge.Main -d {out}_tmp -s -l scala $@
+find {out}_tmp -exec touch -t 198001010000 {{}} \;
+touch -t 198001010000 {manifest}
+{jar} cmf {manifest} {out} -C {out}_tmp .
+rm -rf {out}_tmp
+""".format(
+    java=ctx.file._java.path,
+    out=ctx.outputs.srcjar.path,
+    manifest=ctx.outputs.manifest.path,
+    jar=ctx.file._jar.path,
+    jars=":".join([f.path for f in cjars]),
+  )
+  ctx.action(
+    #TODO we should curate this a bit more deliberately...make sure it is accurate when it all is done
+    inputs=list(ctx.attr.thrift_library.transitive_srcs) +
+        cjars +
+        ctx.files._jdk +
+        ctx.files._scalasdk +
+        [ctx.outputs.manifest, ctx.file._jar],
+    outputs=[ctx.outputs.srcjar], #TODO ijar?
```

Comment:
I don't think so, since this code mechanically depends on the thrift, and if the thrift changes, the ijar is quite likely to change. We don't change logic without changing interface, is my point.